scriptworker/github.py

```diff
@@ -65,3 +76,152 @@ def get_release(self, tag_name):
+        # when the commit hasn't landed on the origin repo. Otherwise, some HTML data is returned - it
+        # represents the branches on which the given revision is present.
+        return html_text != ''
+
+
+def is_github_url(url):
+    """Tell if a given URL matches a Github one.
+
+    Args:
+        url (str): The URL to test. It can be None.
+
+    Returns:
+        bool: False if the URL is not a string or if it doesn't match a Github URL
+
+    """
+    if isinstance(url, str):
+        return url.startswith('https://github.com/')
+    else:
+        return False
+
+
+def extract_github_repo_owner_and_name(url):
+    """Given an URL, return the repo name and who owns it.
+
+    Args:
+        url (str): The URL to the GitHub repository
+
+    Raises:
+        ValueError: on url that aren't from github
+
+    Returns:
+        str, str: the owner of the repository, the repository name
+
+    """
+    _check_github_url_is_supported(url)
+
+    parts = get_parts_of_url_path(url)
+    repo_owner = parts[0]
+    repo_name = parts[1]
+
+    return repo_owner, _strip_trailing_dot_git(repo_name)
+
+
+def extract_github_repo_and_revision_from_source_url(url):
+    """Given an URL, return the repo name and who owns it.
+
+    Args:
+        url (str): The URL to the GitHub repository
+
+    Raises:
+        ValueError: on url that aren't from github or when the revision cannot be extracted
+
+    Returns:
+        str, str: the owner of the repository, the repository name
+
+    """
+    _check_github_url_is_supported(url)
+
+    parts = get_parts_of_url_path(url)
+    repo_name = parts[1]
+    try:
+        revision = parts[3]
+    except IndexError:
+        raise ValueError('Revision cannot be extracted from url: {}'.format(url))
+
+    end_index = url.index(repo_name) + len(repo_name)
+    repo_url = url[:end_index]
+
+    return _strip_trailing_dot_git(repo_url), revision
+
+
+def _strip_trailing_dot_git(url):
+    if url.endswith('.git'):
+        url = url[:-len('.git')]
+    return url
+
+
+def is_github_repo_owner_the_official_one(context, repo_owner):
+    """Given a repo_owner, check if it matches the one configured to be the official one.
+
+    Args:
+        context (scriptworker.context.Context): the scriptworker context.
+        repo_owner (str): the repo_owner to verify
+
+    Raises:
+        scriptworker.exceptions.ConfigError: when no official owner was defined
+
+    Returns:
+        bool: True when ``repo_owner`` matches the one configured to be the official one
+
+    """
+    official_repo_owner = context.config['official_github_repos_owner']
+    if not official_repo_owner:
+        raise ConfigError(
+            'This worker does not have a defined owner for official GitHub repositories. '
+            'Given "official_github_repos_owner": {}'.format(official_repo_owner)
+        )
+
+    return official_repo_owner == repo_owner
+
+
+def _is_git_full_hash(commit_hash):
```

Comment:
Nit: since the input of this function isn't always a commit hash (it may be a branch or a tag), perhaps this should be called "revision" :)

Reply:
Good call. Fixed.